psi4/src/psi4/libcubeprop/csg.cc

```diff
@@ -649,6 +649,33 @@ void CubicScalarGrid::compute_orbitals(std::shared_ptr<Matrix> C, const std::vec
     }
     free_block(v);
 }
+void CubicScalarGrid::compute_difference(std::shared_ptr<Matrix> C, const std::vector<int>& indices,
+                                         const std::string& label, bool square, const std::string& type) {
+    auto C2 = std::make_shared<Matrix>(primary_->nbf(), indices.size());
+    double** Cp = C->pointer();
+    double** C2p = C2->pointer();
+    for (int k = 0; k < indices.size(); k++) {
+        C_DCOPY(primary_->nbf(), &Cp[0][indices[k]], C->colspi()[0], &C2p[0][k], C2->colspi()[0]);
+    }
+    double** v_t = block_matrix(indices.size(), npoints_);
```

Comment:
Can you switch this to a `auto v_t = std::make_shared<Matrix>(indices.size(), npoints_)` and a vector below. Trying to discourage `new/delete/free` in general.